pkg/operators/wasm/testdata/badguest/program.go

```diff
@@ -303,7 +303,7 @@ func gadgetInit() int {
 	assertNonZero(mapDelete(0, invalidStrPtr), "mapDelete: bad handle")
 
 	/* SyscallDeclaration */
-	assertZero(getSyscallDeclaration(invalidStrPtr), "getSyscallDeclaration: bad syscall name pointer")
+	assertNonZero(getSyscallDeclaration(invalidStrPtr, invalidStrPtr), "getSyscallDeclaration: bad syscall name pointer")
```

Comment:
Ideally it should check both conditions separately, something like this (not tested):

```go
	syscallDeclarationPtr := bytesToBufPtr(make([]byte, 32/*FIXME: correct size here*/))
	assertZero(getSyscallDeclaration(stringToBufPtr("open"), syscallDeclarationPtr), "getSyscallDeclaration: good")
	assertNonZero(getSyscallDeclaration(invalidStrPtr, syscallDeclarationPtr), "getSyscallDeclaration: bad syscall name pointer")
	assertNonZero(getSyscallDeclaration(stringToBufPtr("open"), invalidStrPtr), "getSyscallDeclaration: bad syscall decl pointer")
```